core/trigger.class.inc.php

```diff
@@ -113,17 +113,22 @@ public function DoActivate($aContextArgs)
 
 		// Find the related actions
 		$oLinkedActions = $this->Get('action_list');
-		while ($oLink = $oLinkedActions->Fetch())
-		{
-			/** @var \DBObject $oLink */
-			$iActionId = $oLink->Get('action_id');
-			/** @var \Action $oAction */
-			$oAction = MetaModel::GetObject('Action', $iActionId);
-			if ($oAction->IsActive())
-			{
-                $oKPI = new ExecutionKPI();
-				$oAction->DoExecute($this, $aContextArgs);
-                $oKPI->ComputeStatsForExtension($oAction, 'DoExecute');
+		$aActionListOrdered = [];
+		while ($oLink = $oLinkedActions->Fetch()) {
+			$aActionListOrdered[$oLink->Get('order')][] = $oLink;
```

Comment:
```suggestion
			$aActionListOrdered[(int) $oLink->Get('order')][] = $oLink;
```

To ensure that order is always returned as an int, so ksort work even if Get() starts returning strings.